suite-native/device-authentication/src/utils.ts

```diff
@@ -0,0 +1,12 @@
+import { AnyAction } from '@reduxjs/toolkit';
+
+import { UI } from '@trezor/connect';
+
+export const isPinAction = (action: AnyAction) =>
+    action.type === UI.REQUEST_BUTTON &&
+    (action.payload.code === 'ButtonRequest_PinEntry' ||
+        action.payload.code === 'PinMatrixRequestType_Current');
+
+export const isCloseWindowActionOrNonPinAction = (action: AnyAction) =>
+    action.type === UI.CLOSE_UI_WINDOW ||
+    (action.type === UI.REQUEST_BUTTON && !isPinAction(action));
```

Comment:
This looks too complex to me, I would prefer to separate CLOSE_UI_WINDOW condition from request button condition for better readability.

Reply:
Good point. I already have something like this in a cleanup commit that I'm preparing :) 
![image](https://github.com/trezor/trezor-suite/assets/36101761/7801e458-1fdd-4f11-9922-ffabe715f366)

Reply:
Separated. You can check here (I force pushed, sorry):
https://github.com/trezor/trezor-suite/blob/2ad1bf6c528f77d5567665cb766ea0bd007f163b/suite-native/device-authentication/src/utils.ts

Reply:
From my point of view, it didn't improve readability. And it's wrong. So I guess it's really too complex... 🙂 

I would prefer to delete `isCloseWindowActionOrNonPinAction` completely change the matcher in deviceAuthenticationSlice.ts

Reply:
That can be troublesome since this is hacky that we dispatch events from connect. These actions are not created with action creator that would fulfill the conditions to use `isAnyOf` etc. That would result in having 3 different matchers just for ending pin flow and then there would be more for passphrase as well. I would prefer to make these utils more readable instead, if that could work. 

What do you think of this? https://github.com/trezor/trezor-suite/pull/12889/commits/34f49c2959868e0c62f7cb23db62fd3884232352

![image](https://github.com/trezor/trezor-suite/assets/36101761/20820b38-32ba-472b-93fd-1d3483c10407)

Note: I'm considering writing some kind of test for this as well.

Reply:
Updated conditions in slice here: [ff2980a](https://github.com/trezor/trezor-suite/pull/12889/commits/ff2980a3ea0386747c503e51b9591a275032c280)